client/ayon_blender/plugins/publish/validate_absolute_external_data_paths.py

```diff
@@ -0,0 +1,61 @@
+import bpy
+import pyblish.api
+from ayon_core.pipeline import (
+    OptionalPyblishPluginMixin
+)
+from ayon_core.pipeline.publish import (
+    RepairAction,
+    ValidateContentsOrder,
+    PublishValidationError
+)
+
+
+
+class ValidateAbsoluteExternalDataPaths(pyblish.api.InstancePlugin,
+                                        OptionalPyblishPluginMixin):
+    """Validates Absolute External Data Paths
+
+    This validator checks if all external data paths are absolute
+    to ensure the links would not be broke when publishing
+    """
+
+    label = "Validate Absolute External Data Paths"
+    order = ValidateContentsOrder
+    hosts = ["blender"]
+    optional = True
+    actions = [RepairAction]
+
+    @classmethod
+    def get_invalid(cls, instance):
+        invalid = []
+        object_type = type(bpy.data.objects)
+        for attr in dir(bpy.data):
+            collections = getattr(bpy.data, attr)
+            if isinstance(collections, object_type):
+                for data_block in collections:
+                    if hasattr(data_block, "filepath") and data_block.filepath:
+                        if data_block.filepath != bpy.path.abspath(data_block.filepath):
```

Comment:
This would be much less indented (and more readable?) if not using the deep nesting but instead `continue` if something is not what we want.

For example:
```python
        object_type = type(bpy.data.objects)
        for attr in dir(bpy.data):
            collections = getattr(bpy.data, attr)
            if not isinstance(collections, object_type):
                continue
            for data_block in collections:
                if not hasattr(data_block, "filepath") or not data_block.filepath:
                    continue

                if data_block.filepath == bpy.path.abspath(data_block.filepath):
                    continue
                    
                # etc.
```